data/locations/src/main/java/de/mm20/launcher2/locations/providers/openstreetmaps/OsmLocationProvider.kt

```diff
@@ -109,62 +110,68 @@ internal class OsmLocationProvider(
-                }
-            }?.getOrNull()
-
-        val result = awaitAll(
-            // optionally query by "amenity" or "shop" here
-            // if we want to make searching for locations fuzzier
-            // however, this would not account for localized queries like "Bäcker" (shop:bakery)
-            Scope.async { searchByTag("name") },
-            Scope.async { searchByTag("brand") },
-        ).flatMap {
-            it?.let {
-                OsmLocation.fromOverpassResponse(it, context)
-            } ?: emptyList()
-        }
-
-        return result
-            .asSequence()
-            .filter {
-                (!hideUncategorized || (it.category != null)) && it.distanceTo(userLocation) < searchRadiusMeters
-            }
-            .groupBy {
-                it.label.lowercase()
-            }
-            .flatMap { (_, duplicates) ->
-                // deduplicate results with same labels, if
-                // - same category
-                // - distance is less than 100m
-                if (duplicates.size < 2) duplicates
-                else {
-                    val luckyFirst = duplicates.first()
-                    duplicates
-                        .drop(1)
-                        .filter {
-                            it.category != luckyFirst.category ||
-                                    it.distanceTo(luckyFirst) > 100.0
-                        } + luckyFirst
-                }
+            )
+        }?.onFailure {
+            if (it !is HttpException && it !is CancellationException) {
+                Log.e("OsmLocationProvider", "Failed to search for: $query", it)
             }
-            .sortedBy {
-                it.distanceTo(userLocation)
+        }?.getOrNull()?.let {
+            OsmLocation.fromOverpassResponse(it, context)
+        }?.asSequence()?.filter {
+            (!hideUncategorized || (it.category != null)) && it.distanceTo(userLocation) < searchRadiusMeters
+        }?.groupBy {
+            it.label.lowercase()
+        }?.flatMap { (_, duplicates) ->
+            // deduplicate results with same labels, if
+            // - same category
+            // - distance is less than 100m
+            if (duplicates.size < 2) duplicates
+            else {
+                val luckyFirst = duplicates.first()
+                duplicates
+                    .drop(1)
+                    .filter {
+                        it.category != luckyFirst.category ||
+                                it.distanceTo(luckyFirst) > 100.0
+                    } + luckyFirst
             }
-            .take(7)
-            .toImmutableList()
+        }?.sortedBy {
+            it.distanceTo(userLocation)
+        }?.take(7)?.toImmutableList() ?: emptyList()
     }
+
+    private val poiCategories by lazy {
+        R.string::class.java.declaredFields.mapNotNull {
+            if (it.name.startsWith("poi_category_"))
+                it to it.name.removePrefix("poi_category_")
```

Comment:
this is not correct actually. `poi_category_pizza_restaurant` is queried as `amenity=pizza_restaurant`, which is not correct. Ideally, it should be `amenity=restaurant` and `cuisine=pizza`...

Reply:
Added some (restaurant centric for now, of course :) ) special handling for the names of the resource strings.

So, any resouce string with name `poi_category_asdf_restaurant` will query as `cuisine=asdf` and `amenity=restaurant`. This should be extensible to any other categories that combine tags to make sense, with an requirement for strict resource naming.